build.sh

```diff
@@ -548,6 +561,7 @@ __SkipRestore=""
 __SkipNuget=0
 __SkipCoreCLR=0
 __SkipMSCorLib=0
+__SkipRestoreOptData=0
```

Comment:
Similar to what we do for `__SkipGenerateVersion=1`, we should also skip restoring opt data if `__isMSBuildOnNETCoreSupported` is 0. And you would not need to modify the `netci.groovy` then.